### ISSUE TITLE
Provide an option to ignore circular dependencies that are outside the distribution being analyzed

### ISSUE THREAD

**Issue by gforcada** [collective/pipforester#6]
__Use case:__ a distribution (`plone.app.testing`) has a GHA with `pipforester` that fails due to `plone.app.multilingual` and `Products.CMFPlone` having a circular dependency.

It is 100% accurate to report such a circular dependency, but it is outside of `plone.app.testing` control to break that dependency. No changes on the `plone.app.testing` code base will fix that circular dependency.

In such cases, it would be great to have an option in `pipforester` to only _fail_ if the circular dependencies found also involve the given distribution.

i.e. right now in the GHA to check for circular dependencies looks like this:

```
pip install pipdeptree pipforester
pipdeptree -j > forest.json
# Generate a DOT graph with the circular dependencies, if any
pipforester -i forest.json -o forest.dot --cycles
# Report if there are any circular dependencies, i.e. error if there are any
pipforester -i forest.json --check-cycles -o /dev/null
```

If we had another parameter (`-s` ?) then we could update the last command to be:

```
pipforester -s plone.app.testing -i forest.json --check-cycles -o /dev/null
```

And despite reporting that `plone.app.multilingual` and `Products.CMFPlone` have a circular dependency problem, as that one does __not__ involve `plone.app.testing` `pipforester` would return a `0` exit status code, i.e. that all good 👍🏾 

Does that make sense? 🤔

**Comment by mauritsvanrees** [comment]
Yes, I am hoping for something like this as well.

And then we could still have a `circular-full` gh-action check on one central package, like CMFPlone, that does **not** use this potential new option, so we have at one package where it still breaks if there is _any_ circular dependency anywhere in the stack.

**Comment by gforcada** [comment]
Yes, I'm delaying the CMFPlone _conversion_ to `plone/meta` to iron out a bit more the other packages, but we could start with the linting & QA tools first and add tests and circular later as we see that they work on it 👍🏾

**Comment by jensens** [comment]
Probably easy to achieve: After reading of the source graph, remove some given packages and its edges, then proceed as usual. Like after this line

https://github.com/collective/pipforester/blob/f993d7724a8a2ee3a007e836739a7c5c2916cdd9/pipforester/__init__.py#L13

call a new function like so:
```python
deptreedata = remove_nodes_and_their_edges(deptreedata, list_of_nodes_by_name)
```
